Automattic/newspack-plugin
# Adds a filter by Author in the Posts page

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds a filter by Author to the posts page. See issue description for details.

It was inspired by the [Author FIlter](https://wordpress.org/plugins/author-filters/) plugin, but it was simplified and updated:

* No need for pre_get_posts filter
* replace usage of deprecated `who` param in favor of `capability`. Also, make it work with any custom post type
* i18n

Closes #1182 .

### How to test the changes in this Pull Request:

1. Test in a site with more than one author. Ideally with many authors
2. Go to wp-admin
3. Go to Posts
4. Confirm there's a new "Authors" dropdown in the Filters section
6. Test the filter, combined with others, and make sure it works as expected
7. Make sure it works well when you go to the "Mine" tab
8. If possible, test it with custom post types
9. NOTE: I'm concerned that this can degrade the speed of the page for sites with many (hundreds) of users... maybe we could add a limit for now before creating a proper solution?


Test with Co Authors Plus
- Install and activate Co Authors Plus
- Create one or more guest authors (Users -> Guest authors)
- Assign posts to guest authors
- Repeat the tests above and confirm the filter works
- Confirm that the filter does not show up when you select the "Mine" tab

Test disable flag
Add `define( 'NEWSPACK_DISABLE_AUTHORS_FILTER', true );` to your wp-config and make sure the filter is no longer initialized.

### Other information:
